### PR TITLE
Initial add of troubleshooting basics and cloud specific file

### PR DESCRIPTION
**Pull request by mitechie**
This is the start of an improved debugging documentation. I'd like to start with this thought it shows itself to be incomplete as I want to send out a call for folks to send me their troubleshooting tips and tricks and such to build this up with feedback from many folks over time.

### CODE REVIEW

**Comment by jamesbeedy** [comment]
+1

**Comment by evilnick** [comment]
Thanks for the information contained here. Didn't have time to give it a proper review today but will pick it up next week

**Comment by mitechie** [comment]
@evilnick ty, I added some additional notes on logging levels to the PR. I'd appreciate review and I'll start to email folks and see if I can get a collection of troubleshooting notes sent for my curation into these docs going forward.

File: src/en/troubleshooting.md [hunk 1]
```diff
@@ -1,15 +1,214 @@
...
+
+```
+Unit                Workload  Agent       Machine  Public address  Ports  Message
+active-directory/0  waiting   allocating  1                               waiting for machine
+```
+
+However, if I want additional details I can run `juju show-machine 1` to see
+the details about the machine that this unit is meant to be running on.
+
+```
```
Comment (evilnick) [inline src/en/troubleshooting.md]:
```yaml

File: src/en/troubleshooting.md [hunk 2]
```diff
@@ -1,15 +1,214 @@
...
+details all the time. Often Juju status will help provide you a hint that you
+can chase with a more specific command. One of the most important commands is
+`juju show-machine` where an application failing to deploy might be due to the
+machine never getting assigned for Juju's use.
+
+
+For example, if I deploy active-directory into a cloud that does not have
+Windows images available I might get an error showing:
+
+```
```
Comment (evilnick) [inline src/en/troubleshooting.md]:
```bash

File: src/en/troubleshooting.md [hunk 3]
```diff
@@ -1,15 +1,214 @@
...
+
+Here we are given much more clear details as to what Juju is looking for. Make
+sure to leverage the `show-` commands to dive deeper into what's going on.
+juju show-model and show-controller can often provide useful information when
+file bug reports or requesting assistance.
+
+Make sure to look beyond the default information in `juju status`.
+
+
+### Leveraging --debug
```
Comment (evilnick) [inline src/en/troubleshooting.md]: Using the `--debug` option

File: src/en/troubleshooting.md [hunk 4]
```diff
@@ -1,15 +1,214 @@
...
+
+Most commands support a `--debug` option that can be used to get a much more
```
Comment (evilnick) [inline src/en/troubleshooting.md]: ...that can be used to get a more comprehensive idea of ...

File: src/en/troubleshooting.md [hunk 5]
```diff
@@ -1,15 +1,214 @@
...
+specific idea as to what's going on. This is especially true of commands such
+as `juju bootstrap`.
+
+Compare the sample output of `juju deploy --debug`. It includes all of the
+urls that Juju is attempting to reach. These might be of vital important if
```
Comment (evilnick) [inline src/en/troubleshooting.md]: URLs

importance

File: src/en/troubleshooting.md [hunk 6]
```diff
@@ -1,15 +1,214 @@
...
+13:48:43 DEBUG httpbakery client.go:244 client do GET https://api.jujucharms.com/charmstore/v5/xenial/kibana/meta/any?include=id&include=supported-series&include=published {
+13:48:43 DEBUG httpbakery client.go:246 } -> error <nil>
+13:48:44 DEBUG httpbakery client.go:244 client do GET https://api.jujucharms.com/charmstore/v5/kibana/meta/any?include=id&include=supported-series&include=published {
+13:48:45 DEBUG httpbakery client.go:246 } -> error <nil>
+13:48:45 INFO  cmd deploy.go:1000 Located charm "cs:trusty/kibana-15".
+13:48:45 INFO  cmd deploy.go:1001 Deploying charm "cs:trusty/kibana-15".
+13:48:46 DEBUG juju.api monitor.go:35 RPC connection died
+13:48:46 INFO  cmd supercommand.go:465 command finished
+
+```
```
Comment (evilnick) [inline src/en/troubleshooting.md]: The above could potentially do with a few words of explanation

File: src/en/troubleshooting.md [hunk 7]
```diff
@@ -1,15 +1,214 @@
...
+issue. You can find out the current logging level by checking the
+model-config.
+
+```
+juju model-config logging-config
+<root>=WARNING; unit=INFO
+```
+
+
+You can increase the logging level to DEBUG and all the way to TRACE by
```
Comment (evilnick) [inline src/en/troubleshooting.md]: possibly worth a quick mention of the logging levels available

File: src/en/troubleshooting.md [hunk 8]
```diff
@@ -1,15 +1,214 @@
-Title: Juju troubleshooting  
+Title: Juju troubleshooting
```
Comment (pmatulis) [inline src/en/troubleshooting.md]: 'Troubleshooting' ?

File: src/en/troubleshooting.md [hunk 9]
```diff
@@ -1,15 +1,214 @@
...
+        juju-status:
+          current: allocating
+          since: 18 May 2017 13:33:56-04:00
+        machine: "2"
+        public-address: 10.96.51.68
+
+```
+
+Here we're given a lot more details as to the state of things regarding the
+MySQL unit we've deployed. This includes status, timestamps, additional
```
Comment (pmatulis) [inline src/en/troubleshooting.md]: 'A lot more detail is now provided for the deployed MySQL unit.' ?

File: src/en/troubleshooting.md [hunk 10]
```diff
@@ -1,15 +1,214 @@
...
+into what's going on.
+
+
+### Beyond Juju status
+
+Juju status is a summary of what's going on. It cannot provide the full
+details all the time. Often Juju status will help provide you a hint that you
+can chase with a more specific command. One of the most important commands is
+`juju show-machine` where an application failing to deploy might be due to the
+machine never getting assigned for Juju's use.
```
Comment (pmatulis) [inline src/en/troubleshooting.md]: Replace paragraph with:

'The `status` command displays a status summary and is therefore the first command to use while troubleshooting. A second useful command is `juju show-machine`, which provides information on a Juju machine level.' ?

File: src/en/troubleshooting.md [hunk 11]
```diff
@@ -1,15 +1,214 @@
...
+It aggregates logs across the models in a way that allows for seeing
+everything going on as well as diving into the details with advanced filtering
+techniques. See [this Juju logs](./troubleshooting-logs.html) page for
+additional details about the logs Juju keeps.
+
+
+### Increase the logging level
+
+At times it's necessary to increase the logging level to help diagnose an
+issue. You can find out the current logging level by checking the
```
Comment (pmatulis) [inline src/en/troubleshooting.md]: s/find out/discover (or determine, ascertain)

Comment (pmatulis) [inline src/en/troubleshooting.md]: with the `model-config` command:

File: src/en/troubleshooting-clouds.md [hunk 12]
```diff
@@ -0,0 +1,57 @@
...
+
+
+# Cloud specific issues
+
+
+## Common issues setting up clouds
+
+### juju add-credential
+
+The YAML needed to add a credential works with a credential file from another
```
Comment (mitechie) [inline src/en/troubleshooting-clouds.md]: reworded it a bit, let me know if this is more clear.

File: src/en/troubleshooting-clouds.md [hunk 13]
```diff
@@ -0,0 +1,57 @@
...
+system. This means it's expecting a format that includes a root key of
+"credentials:". Often, users pass around a single credential and don't include
+the root key.
+
+For example, a azure.yaml file like so will fail.
```
Comment (mitechie) [inline src/en/troubleshooting-clouds.md]: ty

File: src/en/troubleshooting-clouds.md [hunk 14]
```diff
@@ -0,0 +1,57 @@
...
+
+```
```
Comment (mitechie) [inline src/en/troubleshooting-clouds.md]: ty

File: src/en/troubleshooting-clouds.md [hunk 15]
```diff
@@ -0,0 +1,57 @@
...
+azure:
+  rickscreds:
+      auth-type: service-principal-secret
+      application-id: 1234
+      application-password: secretpassword
+      subscription-id: 12345
+      tenant-id: 12345
+```
+
+Once the root key is added it will succeed.
```
Comment (mitechie) [inline src/en/troubleshooting-clouds.md]: ty

File: src/en/troubleshooting-clouds.md [hunk 16]
```diff
@@ -0,0 +1,57 @@
...
+
+```
```
Comment (mitechie) [inline src/en/troubleshooting-clouds.md]: ty

File: src/en/troubleshooting-clouds.md [hunk 17]
```diff
@@ -0,0 +1,57 @@
...
+credentials:
+    azure:
+      rickstuff:
+          auth-type: service-principal-secret
+          ...
+```
+
+
+## MAAS
```
Comment (mitechie) [inline src/en/troubleshooting-clouds.md]: This is a placeholder for now to encourage users to submit items per my email to the mailing list. I've added the other two per your notes.

File: src/en/troubleshooting.md [hunk 18]
```diff
@@ -1,15 +1,214 @@
...
+When things don't seem to be working the first step is to pull together the
+important information that Juju is providing.
+
+
+### Filtering status
+
+`juju status` supports specifying an application to be shown. In this way,
+`juju status mysql` will only show the status information relevant to that
+application. The default output of `juju status` is a nice tabular format.
+However, sometimes there's additional details in the machine readable output.
```
Comment (mitechie) [inline src/en/troubleshooting.md]: ty

File: src/en/troubleshooting.md [hunk 19]
```diff
@@ -1,15 +1,214 @@
...
+You can see this at any time by passing the flag `--format=yaml` to any status
+command in Juju.
+
+```bash
```
Comment (mitechie) [inline src/en/troubleshooting.md]: updated

File: src/en/troubleshooting.md [hunk 20]
```diff
@@ -1,15 +1,214 @@
...
+    dns-name: 10.96.51.68
+    ip-addresses:
+    - 10.96.51.68
+    instance-id: i-0510eb612038f780a
+    machine-status:
+      current: running
+      message: running
+      since: 18 May 2017 13:34:23-04:00
+    series: xenial
+    harGdware: arch=amd64 cores=1 cpu-power=350 mem=3840M root-disk=8192M availability-zone=eu-west-1a
```
Comment (mitechie) [inline src/en/troubleshooting.md]: guess so...heh

File: src/en/troubleshooting.md [hunk 21]
```diff
@@ -1,15 +1,214 @@
...
+### Beyond Juju status
+
+Juju status is a summary of what's going on. It cannot provide the full
+details all the time. Often Juju status will help provide you a hint that you
+can chase with a more specific command. One of the most important commands is
+`juju show-machine` where an application failing to deploy might be due to the
+machine never getting assigned for Juju's use.
+
+
+For example, if I deploy active-directory into a cloud that does not have
```
Comment (mitechie) [inline src/en/troubleshooting.md]: reworded ty, it was awkward.

File: src/en/troubleshooting.md (hunk 2 above)
Comment (mitechie) [inline src/en/troubleshooting.md]: ty

File: src/en/troubleshooting.md [hunk 22]
```diff
@@ -1,15 +1,214 @@
...
+    machine-status:
+      current: provisioning error
+      message: no "win2012" images in Region1 with arches [amd64]
+      since: 18 May 2017 13:26:55-04:00
+    series: win2012
+```
+
+Here we are given much more clear details as to what Juju is looking for. Make
+sure to leverage the `show-` commands to dive deeper into what's going on.
+juju show-model and show-controller can often provide useful information when
```
Comment (mitechie) [inline src/en/troubleshooting.md]: ty

**Comment by mitechie** [comment]
@evilnick and @pmautils PTAL

**Comment by mitechie** [comment]
doh @pmatulis that is ^

**Comment by evilnick** [comment]
I think there are still some issues around the wording in places, I'm not sure what happened there. I think though that the easiest/least time-consuming way to proceed with this is to merge it and then we will fix it up.

**Pull request merged**